src/wasm-type.h

```diff
@@ -453,31 +455,6 @@ class RecGroup {
   HeapType operator[](size_t i) const { return *Iterator{{this, i}}; }
 };
 
-using TypeList = std::vector<Type>;
-
-// Passed by reference rather than by value because it can own an unbounded
-// amount of data.
-struct Tuple {
-  TypeList types;
-  Tuple() : types() {}
-  Tuple(std::initializer_list<Type> types) : types(types) { validate(); }
-  Tuple(const TypeList& types) : types(types) { validate(); }
-  Tuple(TypeList&& types) : types(std::move(types)) { validate(); }
-
-  bool operator==(const Tuple& other) const { return types == other.types; }
-  bool operator!=(const Tuple& other) const { return !(*this == other); }
-  std::string toString() const;
-
-private:
-  void validate() {
-#ifndef NDEBUG
-    for (auto type : types) {
-      assert(type.isSingle());
```

Comment:
Do we have this logic already in the validator? If not it might be good to add.

Reply:
Oh right, I definitely need to replace this with something...